include/seqan3/alignment/pairwise/align_result_selector.hpp

```diff
@@ -95,21 +74,15 @@ struct align_result_selector
         }
         else if constexpr (configuration_t::template exists<align_cfg::result<with_alignment_type, score_type>>())
         {
-            // Due to an error with gcc8 we define these types beforehand.
-            using first_gapped_seq_type = gapped<value_type_t<first_range_t>>;
-            using second_gapped_seq_type = gapped<value_type_t<second_range_t>>;
-
-            // We use vectors of gapped sequence if the gap decorator cannot be used.
-            using fallback_t = std::tuple<std::vector<first_gapped_seq_type>, std::vector<second_gapped_seq_type>>;
-
-            // If the ranges are RandomAccess and Sized we can use the Gap Decorator, otherwise fallback_t.
-            using decorator_t = alignment_type<first_range_t, second_range_t>;
+            // TODO: type `(first|second)_range_t &` is a hack to make the sequences viewable_ranges
```

Comment:
if 
https://github.com/seqan/seqan3/blob/034810f0c16069321670f84be6ca9cb52a8f88a7/include/seqan3/alignment/pairwise/alignment_algorithm.hpp#L182

was `range_reference_type`, you could require `viewable_range_ here?

Can this somehow backfire because you'll later use this type to create a reference to something that is, e.g., temporary?

Reply:
This might be the way to solve this, we use `value_type` on multiple occurrences in the alignment code, so we might want to fix this on multiple occasions, but maybe in another PR.

Reply:
but then the above maybe shouldn't be part of this PR at all? It feels like we introduce a fragile workaround while we know there is more to fix.

Reply:
I think here are two issues. The first one is that the call of this metafunction happens on multiple places possibly making it possible to use wrong types in either place and causing some hard to track compiler errors.
https://github.com/seqan/seqan3/blob/034810f0c16069321670f84be6ca9cb52a8f88a7/include/seqan3/alignment/pairwise/alignment_configurator.hpp#L258

So the first ting we need to do is to use the result type that is defined in the configurator and pass it to the alignment algorithm object, so we avoid redefinition of the same type with possibly wrong types.

Second, the input sequence is either transformed right at the beginning of align_pairwise to always return a pair of views, e.g. like `input_range | transform([](auto && pair) { return pair{tpl.first | type_erase, pair.second | type_erase } } )` and use this throughout the alignment implementation. In this case we assume everything is a view and we don't have to worry anymore about the correct references of the types. 
Not sure if this has any runtime implications. 
Or we get the types correctly in the first place like this:
```cpp
// assuming std::ranges::viewable_range<input_range_t> is true
using std::get;
using ref_t = std::ranges::range_reference_t<input_range_t>;
using seq1_t = decltype(get<0>(std::declval<ref_t &&>()));
using seq2_t = decltype(get<1>(std::declval<ref_t &&>()));
```
Here the reference type should be correctly encoded in the `seqi_t` types and can be used inside of the meta functions.